bazel-build.el

```diff
@@ -60,23 +66,20 @@ FILE-NAME is the file-name of the current buffer.  WORKSPACE-ROOT is
 the root of the Bazel workspace."
   (concat "//" (directory-file-name (bazel-build--extract-package-name file-name workspace-root))))
 
-(defun bazel-build--extract-package-name (file-name workspace-root)
-  "Extract the nearest Bazel package for FILE-NAME under WORKSPACE-ROOT."
-  (let ((nearest-build-file (or (bazel-build--find-nearest-build-package file-name) workspace-root)))
-    (file-relative-name nearest-build-file workspace-root)))
-
-(defun bazel-build--find-nearest-build-package (file-name)
-  "Return nearest Bazel build package to FILE-NAME.
-This is the directory containing the first BUILD file up the directory
-from FILE-NAME."
-  (let ((nearest-build-package
-         (cl-some (lambda (build-name) (locate-dominating-file file-name build-name)) bazel-build--recognized-build-file-names)))
-    (if nearest-build-package (expand-file-name nearest-build-package))))
-
 (defconst bazel-build--recognized-build-file-names
   '("BUILD.bazel" "BUILD")
   "Names files Bazel will look in to find build rules.")
 
+(defun bazel-build--extract-package-name (file-name workspace-root)
+  "Extract the nearest Bazel package for FILE-NAME under WORKSPACE-ROOT."
+  (file-relative-name
+   (expand-file-name
+	(or
+	 (cl-some
+	  (lambda (build-name) (locate-dominating-file file-name build-name)) bazel-build--recognized-build-file-names)
+	 workspace-root))
```

Comment:
I don't think this is correct. A directory without a BUILD file is never a package, so you should probably rather signal an error (or return nil) here.